mlir/include/mlir/Interfaces/MemorySlotInterfaces.td

```diff
@@ -83,11 +83,7 @@ def PromotableAllocationOpInterface
 def PromotableMemOpInterface : OpInterface<"PromotableMemOpInterface"> {
   let description = [{
     Describes an operation that can load from memory slots and/or store
-    to memory slots. Loads and stores must be of whole values of the same
-    type as the slot itself.
-
-    For a memory operation on a slot to be valid, it must operate on the slot
-    pointer *only as a pointer to an element of the type of the slot*.
+    to memory slots.
```

Comment:
Doesn't this remain true in the context of the `PromotableMemOpInterface` used by the Mem2Reg pass?

Reply:
It's not yet fully clear if this requirement will remain valid for LLVM dialect. I can remove this change and adapt it when it shows to no longer hold.

Reply:
After some more though, this is really no longer needed. Changing the LLVM load and store interface implementations to support loading an smaller type from a larger slot should be easy enough.
We are in fact considering such changes, as the type consistency pass showed to produce substantial performance degradations for parts of our benchmark set.

Reply:
What about loading larger types from smaller slots?

Reply:
This is not supported, as this would break things on multiple levels. See the stacked PR for the constrains added specifically in SROA. The additional steps for Mem2Reg will be added in subsequent patches.

Reply:
I don’t understand, I would say it is obviously not supported, hence why the original documentation was there. I’m not sure where else the constraints on bounding are documented if you remove this.

Reply:
I can clarify in the comment  that only in-bounds accesses should be considered valid.